File: 3764-maximum-sum-with-at-most-k-elements/maximum-sum-with-at-most-k-elements.cpp
class Solution {
public:
    long long maxSum(vector<vector<int>>& grid, vector<int>& limits, int k) {
        vector<int> select;
        for(int i = 0; i < grid.size();i++){
            int times  = limits[i];
            sort(grid[i].begin(), grid[i].end());
            for(int j = grid[i].size() - 1; times > 0 ; j--){
                select.push_back(grid[i][j]);
                times --;
            }
        }
        if(select.size() == k){
            long long sum = 0;
            for(int i = 0; i < select.size();i++){
                sum += select[i];
            }
            return sum;
        }
        sort(select.begin(), select.end());
        long long sum = 0;
        for(int i = select.size() - 1; k > 0; i--){
            sum += select[i];
            k--;
        }
        return sum;
    }
};

File: 3764-maximum-sum-with-at-most-k-elements/maximum-sum-with-at-most-k-elements.py
class Solution:
    def maxSum(self, grid: List[List[int]], limits: List[int], k: int) -> int:
        candidates = []

        # Extract the top elements from each row
        for i in range(len(grid)):
            row = sorted(grid[i], reverse=True)  # Sort row in descending order
            candidates.extend(row[:limits[i]])  # Take at most 'limits[i]' elements

        # Get the k largest elements
        return sum(heapq.nlargest(k, candidates)) if k > 0 else 0
        